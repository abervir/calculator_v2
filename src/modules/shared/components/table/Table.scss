@import '../../../../assets/styles/variables.scss';

.table {
  width: 100%;
  table-layout: fixed;

  tr {
    td {
      border: $border;

      &:first-child {
        width: 70px;
        text-align: center;
      }
    }

    &:hover {
      cursor: pointer;

      td {
        background: $border-color;
      }
    }
  }
}
