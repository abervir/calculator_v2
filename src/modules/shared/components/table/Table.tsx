import React from 'react';

import './Table.scss';

interface Props {
  data: any[];
  render: (item: any) => JSX.Element;
  rowKeyProperty: string;
}

export const Table = (props: Props) => {
  const { data, rowKeyProperty, render } = props;

  const rowsRender = (): JSX.Element[] => {
    return data.map((item, index) => {
      const rowId = item[rowKeyProperty];

      return (
        <tr key={rowId}>
          <td>{index + 1}</td>
          {render(item)}
        </tr>
      );
    });
  };
  return (
    <table className='table'>
      <tbody>{rowsRender()}</tbody>
    </table>
  );
};
