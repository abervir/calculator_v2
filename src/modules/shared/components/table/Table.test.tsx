/* eslint-disable jest/no-conditional-expect */
import { shallow } from 'enzyme';
import React from 'react';

import { Table } from './Table';

describe('Table', () => {
  it('should render component', () => {
    const data = [
      {
        name: 'test',
        value: 5,
      },
    ];

    expect(
      shallow(
        <Table
          data={data}
          rowKeyProperty='name'
          render={(item) => (
            <>
              <td>{item.name}</td>
              <td>{item.value}</td>
            </>
          )}
        />
      )
    ).toMatchSnapshot();
  });
});
