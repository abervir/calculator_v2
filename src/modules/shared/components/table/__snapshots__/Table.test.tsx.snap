// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render component 1`] = `
<table
  className="table"
>
  <tbody>
    <tr
      key="test"
    >
      <td>
        1
      </td>
      <td>
        test
      </td>
      <td>
        5
      </td>
    </tr>
  </tbody>
</table>
`;
